### PR TITLE
Added calendar support

### PR DESCRIPTION
**Pull request by galeone**
# Description

The library only supports the traditional heatmap. echarts supports also the [calendar variation](https://echarts.apache.org/examples/en/editor.html?c=calendar-heatmap).

I added the support for the calendar. It's now possible to add multiple series and multiple calendars

---
# Type of change

- [X] New feature (Non-breaking change which adds functionality)

---
# Examples

This is how I use it go potentially generate a set of multiple calendars per year. After the code you can find the generated calendar (only for 1 year).

```go
	var dailyStepsPerYear map[int][]opts.HeatMapData = make(map[int][]opts.HeatMapData)
	var maxSteps int = 0
	for _, dayData := range all {
		if dayData == nil || dayData.Steps == nil {
			continue
		}
		step := int(dayData.Steps.Value)
		if step > maxSteps {
			maxSteps = step
		}
		// format date to YYYY-MM-DD
		value := [2]interface{}{dayData.Date.Format(time.DateOnly), step}
		year := dayData.Date.Year()
		dailyStepsPerYear[year] = append(dailyStepsPerYear[year], opts.HeatMapData{Value: value, Name: dayData.Date.Format(time.DateOnly)})
	}
	chart := charts.NewHeatMap()

	chart.SetGlobalOptions(
		charts.WithTitleOpts(opts.Title{
			Title: "Daily Steps Count",`
			Top:   "30",
			Left:  "center",
		}),
		charts.WithInitializationOpts(opts.Initialization{
			Theme: "dark",
		}),
		charts.WithTooltipOpts(opts.Tooltip{
			Trigger: "item",
			Show:    true,
		}),
		charts.WithVisualMapOpts(
			opts.VisualMap{
				Type:   "piecewise",
				Min:    0,
				Max:    float32(maxSteps),
				Show:   true,
				Orient: "horizontal",
				Left:   "center",
				Top:    "65",
				TextStyle: &opts.TextStyle{
					Color: "white",
				},
			}),
		charts.WithLegendOpts(opts.Legend{
			Show: false,
		}),
	)

	years := make([]int, 0, len(dailyStepsPerYear))
	for k := range dailyStepsPerYear {
		years = append(years, k)
	}

	sort.Ints(years)

	verticalOffset := 120
	for id, year := range years {
		chart.AddSeries("Daily Steps", dailyStepsPerYear[year],
			charts.WithCoordinateSystem("calendar"),
			charts.WithCalendarIndex(id),
		)
		chart.AddCalendar(&opts.Calendar{
			Orient:   "horizontal",
			Silent:   false,
			Range:    []float32{float32(year)},
			Top:      fmt.Sprintf("%d", 120+id*verticalOffset),
			Left:     "30",
			Right:    "30",
			CellSize: "15",
			ItemStyle: &opts.ItemStyle{
				BorderWidth: 0.5,
			},
			YearLabel: &opts.CalendarLabel{
				Show: true,
			},
			DayLabel: &opts.CalendarLabel{
				Show:  true,
				Color: "white",
			},
			MonthLabel: &opts.CalendarLabel{
				Show:  true,
				Color: "white",
			},
		})
	}

	return chart
}
```

image

### CODE REVIEW

File: types/orderedset_test.go [hunk 1]
```diff
@@ -1,8 +1,9 @@
 package types
 
 import (
-	"github.com/stretchr/testify/assert"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
 )
```
Comment (Koooooo-7) [inline types/orderedset_test.go:7]: Plz remove the unnecessary format changes.

**Review by Koooooo-7 — CHANGES_REQUESTED** [review]

**Comment by Koooooo-7** [comment]
Hi @galeone , thx for your awesome chart contributions and the nice example !
Could plz add the example in the [examples](https://github.com/go-echarts/examples)? thx in advance.

File: types/orderedset_test.go (hunk 1 above)
Comment (galeone) [reply types/orderedset_test.go:7]: It was formatted running your script `build.sh format`. I guess something changed in the Go toolchain and this is a new rule.

Should I discard it anyway?

Comment (Koooooo-7) [reply types/orderedset_test.go:7]: > It was formatted running your script `build.sh format`. I guess something changed in the Go toolchain and this is a new rule.
> 
> Should I discard it anyway?

Yep. plz discard it , I gonna check the `build.sh format` later. thx for the inconvenient. 😅

Comment (galeone) [reply types/orderedset_test.go]: done

**Comment by galeone** [comment]
> Hi @galeone , thx for your awesome chart contributions and the nice example ! Could plz add the example in the [examples](https://github.com/go-echarts/examples)? thx in advance.

Right now I don't have a lot of time for doing it (this is work I'm doing for a personal project). I'll try to contribute to the example section during this weekend. If in the meantime you can merge this addition it would be great

**Comment by Koooooo-7** [comment]
> > Hi @galeone , thx for your awesome chart contributions and the nice example ! Could plz add the example in the [examples](https://github.com/go-echarts/examples)? thx in advance.
> 
> Right now I don't have a lot of time for doing it (this is work I'm doing for a personal project). I'll try to contribute to the example section during this weekend. If in the meantime you can merge this addition it would be great

I see, take your time, thats fine.

**Review by Koooooo-7 — APPROVED** [review]
LGTM.

**Pull request merged**